### PR TITLE
docs: design concepts page

### PR DESCRIPTION
**Pull request by Ulexus**
Signed-off-by: Seán C McCord <ulexus@gmail.com>

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/5014)

### CODE REVIEW

File: website/content/docs/v0.15/Learn More/concepts.md [hunk 1]
```diff
@@ -3,10 +3,146 @@ title: "Concepts"
...
+a single node one and a two-node one.
+Needless to say, this was an unhappy arrangement.
+
+Fortunately, `init` nodes are gone, but that means that the critical operation
+of forming a cluster is a manual process.
+It's an _easy_ process, consisting of a single API call, but it can be a
+confusing one, until you understand what it does.
+
+Every new cluster must be bootstrapped exactly and only once.
+This means you do NOT bootstrap each node in a cluster, not even each bootstrap
```
Comment (andrewrynhard) [inline website/content/docs/v0.15/Learn More/concepts.md:127]: "not even each bootstrap node" sounds strange to me. Typo or am I just not reading it right?

**Review by andrewrynhard — APPROVED** [review]
LGTM so far! We really need this doc. is the TODO for another PR or a TODO you wanted for this PR?

File: website/content/docs/v0.15/Learn More/concepts.md (hunk 1 above)
Comment (Ulexus) [reply website/content/docs/v0.15/Learn More/concepts.md:127]: Hah, that second should read "control plane", not bootstrap.

**Comment by Ulexus** [comment]
Ohh.  Storage.  Storage must be mentioned here, too

**Comment by Ulexus** [comment]
> LGTM so far! We really need this doc. is the TODO for another PR or a TODO you wanted for this PR?

Yeah, of i can come up with a good way to check it, i want to add it.  If i don't, there will have to bea follow-up pr.

File: website/content/docs/v0.15/Learn More/concepts.md [hunk 2]
```diff
@@ -3,10 +3,146 @@ title: "Concepts"
...
+This means fail-overs are handled easily, often, and usually without any notice
+of operators.
+This _also_ means that the operational architecture is fundamentally different.
+
+Properly managed (which Talos Linux does), `etcd` should never have split brain
+and should never encounter noticeable down time.
+In order to do this, though, `etcd` maintains the concept of "membership" and of
+"quorum".
+In order to perform _any_ operation, read _or_ write, the database requires
+quorum to be achieved.
```
Comment (Ulexus) [inline website/content/docs/v0.15/Learn More/concepts.md:79]: Achieved->sustained

Comment (Ulexus) [reply website/content/docs/v0.15/Learn More/concepts.md:79]: Maybe maintained

File: website/content/docs/v0.15/Learn More/concepts.md [hunk 3]
```diff
@@ -3,10 +3,146 @@ title: "Concepts"
...
+
+This design means, however, that having an incorrect number of members can be
+devastating.
+Having only two controlplane nodes, for instance, is mostly _worse_ than having
+only one, because if _either_ goes down, your entire database will lock.
+You would be better off just making periodic snapshots of the data and restoring
+it when necessary.
+
+Another common situation occurs when replacing controlplane nodes.
+If you have three controlplane nodes and replace one, you will not have three
```
Comment (Ulexus) [inline website/content/docs/v0.15/Learn More/concepts.md:98]: Not->now

**Pull request merged**